test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2021  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from numpy import *
+from essentia_test import *
+
+class TestPitchSalienceFunction(TestCase):
+  
+    def testInvalidParam(self):
+        self.assertConfigureFails(PitchSalienceFunction(), {'binResolution': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'harmonicWeight': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'harmonicWeight': 2})
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeCompression': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeCompression': 2})        
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeThreshold': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'numberHarmonics': 0})
+        self.assertConfigureFails(PitchSalienceFunction(), {'numberHarmonics': -1})        
+        self.assertConfigureFails(PitchSalienceFunction(), {'referenceFrequency': -1})
+
+    def testZero(self):
+        freqs = zeros(1024)
+        mags  = zeros(1024)
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunction()(freqs, mags))
+
+    def testEmpty(self): 
+        self.assertEqualVector(PitchSalienceFunction()([], []), zeros(600))
+
+    def testOne(self):        
+        self.assertEqualVector(PitchSalienceFunction()(ones(1024), ones(1024)), zeros(600))
```

Comment:
This test also doesn't make much sense (it gives 1024 peaks at 1 Hz below the minimum frequency of 55 Hz, with the amplitude 1). 

Better tests would be:
- Provide a single input peak with a unit magnitude at the reference frequency, and validate that the output salience function has only one non-zero element at the first bin.
- Provide multiple duplicate peaks at the reference frequency. 
- Provide a single input peak below the reference frequency, so that the result is an empty pitch salience function
- How does a single input peak above the maximum bin affect the salience function? You would see the activations at the subharmonics of the input frequency, that fall within the 6000-cent frequency range.